src/main/java/service/converter/DataConverterImpl.java

```diff
@@ -0,0 +1,39 @@
+package service.converter;
+
+import java.util.ArrayList;
+import java.util.List;
+import model.FruitTransaction;
+
+public class DataConverterImpl implements DataConverter {
+    private static final String SEPARATOR = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int HEADER_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        List<FruitTransaction> fruitTransactionList = new ArrayList<>();
+        inputReport.remove(HEADER_INDEX);
+
+        for (String input : inputReport) {
+            FruitTransaction fruitTransaction = new FruitTransaction();
+            String[] inputSplitter = input.split(SEPARATOR);
+            fruitTransaction.setOperation(findEnumValue(inputSplitter[OPERATION_INDEX]));
+            fruitTransaction.setFruit(inputSplitter[FRUIT_INDEX]);
+            fruitTransaction.setQuantity(Integer.parseInt(inputSplitter[QUANTITY_INDEX]));
+            fruitTransactionList.add(fruitTransaction);
+        }
+
+        return fruitTransactionList;
+    }
+
+    public static FruitTransaction.Operation findEnumValue(String operationCode) {
+        for (FruitTransaction.Operation operation : FruitTransaction.Operation.values()) {
+            if (operation.getCode().equals(operationCode)) {
+                return operation;
+            }
+        }
+        throw new IllegalArgumentException("No operations with: " + operationCode);
+    }
+}
```

Comment:
It seems like the `DataConverterImpl` class is responsible for both parsing data and finding enum values based on strings. Consider separating these responsibilities to adhere to the Single Responsibility Principle (one of the SOLID principles). You could have a dedicated enum parser or a method within the enum itself that handles the conversion from a string to an enum value.